File: Daily/PY/Leetcode1877-数组中最大数对和的最小值.py
from typing import List

class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        l,r = 0,len(nums) - 1
        mm = 0
        while l < r:
            mm = max(nums[l] + nums[r],mm)
            l += 1
            r -= 1
        return mm

File: Leetcode/Rust/Leetcode1877-数组中最大数对和的最小值.rs
impl Solution {
    pub fn min_pair_sum(mut nums: Vec<i32>) -> i32 {
        nums.sort_unstable();
        let mut mm = 0;
        let (mut l,mut r) = (0,nums.len() - 1);
        while l < r{
            mm = mm.max(nums[l] + nums[r]);
            r -= 1;
            l += 1;
        }
        mm as i32
    }
}